### PR TITLE
Creating Contributing file

### PR DESCRIPTION
**Pull request by Yuval-Vino**
Creating Contributing.md file that covers all PR's/Issues guidelines.
Those guidelines will direct us throughout the project and help us maintain a uniform approach
#1 relevant issue

### CODE REVIEW

File: Contributing.md
Comment (jeniaSakirko) [inline Contributing.md:5]: I think we should ask the contributor to open an issue and make sure he adding link in the PR

File: Contributing.md
Comment (jeniaSakirko) [inline Contributing.md:7]: Maybe we better to create a CHANGELOG and ask the contributor to update that file and of course update the README only when relevant

File: Contributing.md
Comment (jeniaSakirko) [inline Contributing.md:10]: Do we have such restriction that at least two need to approve? in not lets make sure we adding that as well

File: Contributing.md
Comment (Yuval-Vino) [reply Contributing.md:5]: good idea, I will commit changes

File: Contributing.md
Comment (Yuval-Vino) [reply Contributing.md:10]: Yes it is indicated in the exercise file

File: Contributing.md
Comment (Yuval-Vino) [reply Contributing.md:7]: that's sounds better, lets see what others think

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:8]: I think "issue a pull request" needs a section of its own - how should a PR look? what should it contain? how should the title\description look? linking an issue to a PR (this very PR needs that as well)

File: Contributing.md
Comment (Omeramsc) [reply Contributing.md:7]: Changes to the README should probably exist only when there are changes relevant to the essence of the project, or to the ways to deploy it.

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:11]: I'm missing a possible section about how should a good code-review look.
also, please note (and add it to the guide) that PRs that do not pass the future CI\tests\checks or have conflicts - may not get reviewed until those are solved.

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:14]: maybe "choose an issue" or "what can I work on"

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:18]: only relevant to a problem with the docs? the issues are used for every change\bug\feature\feat needed.

**Review by Omeramsc — CHANGES_REQUESTED** [review]

File: Contributing.md
Comment (jeniaSakirko) [reply Contributing.md:5]: following @Omeramsc comment [https://github.com/redhat-beyond/Beyond-07-team-1/pull/2#discussion_r816754923](here) I think this is the expected

**Review by jeniaSakirko — APPROVED** [review]
LGTM

**Review by Omeramsc — APPROVED** [review]
LGTM

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:8]: otherwise, it may not get reviewed.

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:17]: need at least +2 to merge

File: Contributing.md
Comment (ormergi) [inline Contributing.md:3]: nit: Consider add link to the code of conduct

File: Contributing.md
Comment (ormergi) [reply Contributing.md:8]: Agree

File: Contributing.md
Comment (ormergi) [inline Contributing.md:11]: Note that it is expected that when this PR is merged it is expected that flake8 does setup up.
Otherwise it may be misleading for the contributors.
There are two options here:
- Wait for #10 to merge
- Omit the part about flake8, and add it at #10 PR.

File: Contributing.md
Comment (ormergi) [inline Contributing.md:7]: Please move the note out of the parentheses _(Changes to README file should be commited **ONLY** when there are changes relevant to the essence of the project, or the ways to deploy it!)_.

File: Contributing.md
Comment (ormergi) [inline Contributing.md:12]: Please consider adding a section about commit messages (explain the why and not what has been changed, new line between title and body, must have sign-off, commit should not include out of scope changes ("atomic" commits).
I recommend this[ blog post](https://cbea.ms/git-commit/) to get sense of how a commit should look like :slightly_smiling_face:

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi Yuval, thanks for adding contribution file it will surly improve productivity.

Please squash all commit to the first one, there is no need for intermediate commits (e.g: resolving code-review changes).

**Review by ErezCohenn — CHANGES_REQUESTED** [review]

File: Contributing.md
Comment (Yuval-Vino) [reply Contributing.md]: > There is a missing link to Labels: should add that link: https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels

fixed

File: Contributing.md
Comment (Yuval-Vino) [reply Contributing.md:73]: > I think we should add instructions to the content that is written in the issue, maybe something like that:
> 
> * **Title:**
>   Must be clear, concise, and accurately describe what the issue is about.
> * **Description**
>   Must include a clear explanation about the reasons for doing the work. Preferably in the form of a user story
>   that clearly mentions the stakeholders involved and their wishes.
> * **Acceptance criteria**
>   Must be a list of measurable achievements that indicate the work in the issue is done.

Added to file

**Review by ErezCohenn — APPROVED** [review]
LGTM !

**Review by tamirmatok — APPROVED** [review]

**Review by OfirMatasas — APPROVED** [review]
The file looks good to me.
Approved!

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:19]: You don't have to ask the second reviewer, just someone with permissions to do so :) maintainers are usually a good choice.

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:21]: use example instead of EX
might want to even change it to a kind-of template.

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:31]: You can link to the conventional commit guide instead of quoting from it. just make sure you make it easy to understand this project is following this convention and it has to be followed through.
https://www.conventionalcommits.org/en/v1.0.0/

**Comment by Omeramsc** [comment]
Overall a very good job! I left some more comments I think should be addressed

File: Contributing.md
Comment (Yuval-Vino) [reply Contributing.md]: > You don't have to ask the second reviewer, just someone with permissions to do so :) maintainers are usually a good choice.

Fixed

**Comment by Yuval-Vino** [comment]
> Overall a very good job! I left some more comments I think should be addressed

I address them all, if there is anything else  please let me know,

**Comment by Omeramsc** [comment]
> > Overall a very good job! I left some more comments I think should be addressed
> 
> I address them all, if there is anything else please let me know,

Yeah, add a point to make sure the commits are signed-off
you do that by adding the `signoff` option to a git commit: `git commit --signoff`

**Comment by Yuval-Vino** [comment]
> > > Overall a very good job! I left some more comments I think should be addressed
> > 
> > 
> > I address them all, if there is anything else please let me know,
> 
> Yeah, add a point to make sure the commits are signed-off you do that by adding the `signoff` option to a git commit: `git commit --signoff`

Done

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:32]: You should move the Commiting section up; you commit before you open a PR :)

File: Contributing.md
Comment (Omeramsc) [inline Contributing.md:50]: maybe merge this section and the issue-reporting guidelines

**Comment by Omeramsc** [comment]
left 2 small comments, let's merge it soon :)

**Comment by Yuval-Vino** [comment]
> left 2 small comments, let's merge it soon :)

Fixed

**Review by Omeramsc — APPROVED** [review]
LGTM, great job!

**Pull request merged**
